src/Compiler/Checking/infos.fs

```diff
@@ -1562,6 +1579,16 @@ type ILPropInfo =
     /// Indicates if the IL property has a 'set' method
     member x.HasSetter = Option.isSome x.RawMetadata.SetMethod
 
+    /// Indidcates whether IL property has an init-only setter (i.e. has the `System.Runtime.CompilerServices.IsExternalInit` modifer)
+    member x.IsSetterInitOnly =
+        assert x.HasSetter
+        if x.HasSetter then
+            match x.SetterMethod.ILMethodRef.ReturnType with
+            | ILType.Modified(_, cls, _) -> cls.FullName = "System.Runtime.CompilerServices.IsExternalInit"
```

Comment:
This code is duplicated in `HasExternalInit` maybe there could be a helper function for it?